spec/docs/termmap.md

```diff
@@ -1,163 +1,99 @@
 which is either an [IRI]() or [Literal](), or
 a [Predicate Map](),
-which should generate an [RDF Term]() which is an [IRI](),
-or to generate an [IRI]() for an [Object Map](),
-then the default [Term Type]() needs to be overwritten.
+which should generate an [=RDF term=] which is an [IRI](),
+or to generate an [IRI]() for an [=object map=],
+the default [Term Type]() needs to be overwritten.
 
-### From a Template (rml:template)
+### From a Template (`rml:template`)
 
-A [template-valued term map]() is represented by a resource
+A <dfn>template-valued term map</dfn> is a [=template-valued expression map=], and is thus represented by a resource
 that has exactly one `rml:template` property.
-The value of the `rml:template` property MUST be a valid [string template]().
-
-A _**string template**_ is a format string that can be used
-to build strings from multiple components.
-It can reference [references]() by enclosing them in curly braces (“{” and “}”).
-The following syntax rules apply to valid string templates:
-
-* Pairs of unescaped curly braces MUST enclose valid references.
-* Curly braces that do not enclose references MUST be escaped by a backslash character (“\”).
-This also applies to curly braces within references.
-* Backslash characters (“\”) MUST be escaped by preceding them with another backslash character, yielding “\\”. This also applies to backslashes within column names.
-* There SHOULD be at least one pair of unescaped curly braces.
-* If a template contains multiple pairs of unescaped curly braces,
-then any pair SHOULD be separated from the next one by a _**safe separator**_.
-This is any character or string
-that does not occur anywhere in any of the data values of either reference;
-or in the [IRI-safe versions]() of the data values,
-if the [term type]() is `rml:IRI`.
-
-The _**template value**_ of the [term map]() for a given logical iteration
-is determined as follows:
-
-1. Let result be the [template string]()
-2. For each pair of unescaped curly braces in result:
-    1. Let value be the data value of the column whose name is enclosed in the curly braces
-    2. If value is NULL, then return NULL
-    3. Let value be the [natural RDF lexical form]() corresponding to value
-        If the [term type]() is `rml:IRI`,
-        then replace the pair of curly braces with an [IRI-safe version]() of value;
-        otherwise, replace the pair of curly braces with value
-3. Return result
-
-The IRI-safe version of a string is obtained by applying the following transformation
+
+If the [=term type=] of the [=template-valued term map=] is `rml:IRI`, then a [=reference value transforming function=] should be applied during the evaluation of the [=template expression=]. The [=reference value transforming function=] should transform a [=reference value=] into an [=IRI-safe version=] of that value.
+
+The <dfn data-lt="IRI-safe">IRI-safe version</dfn> of a string is obtained by applying the following transformation
```

Comment:
I would like to see a good algorithm here, using an URI encoder in programming languages is super expensive and we need a lot of IRIs. Maybe we can have a nicer thing here in the spec?

Reply:
I mostly refer to what we have in an Expression map to fill in a template. There's also an 'algorithm' there for filling in templates. However, this was mostly an issue in the old spec, by separating the Expression and Term Maps, it is 'solved' I think because you have a resolved expression in your Term Map and if it is an IRI, you apply the safety thing on the full IRI.

However, following that, it conflicts with what is said above: apply an IRI safe version of every reference value (which is the job of an Expression, not Term).